src/pages/on-chain-governance/_components/ConfirmVoteDialog.tsx

```diff
@@ -46,14 +48,10 @@ export function ConfirmVoteDialog({
   const [userSelectedVote, setUserSelectedVote] = useState<VoteDecision>();
 
   function closeStates() {
-    if (!isConfirmDetailsClicked) {
-      setIsMasterNodeConfirmClicked(false);
-      setHasUserSelectedVote(false);
-      setIsLoading(false);
-    }
-    if (!isMasterNodeConfirmClicked) {
-      setMasterNodeID(localStorage.getItem("masternodeID") ?? "");
-    }
+    setVoteStage(VoteStages.VoteProposal);
+    setUserSelectedVote(userConfirmedSelectedVote);
+    setIsLoading(false);
+    setMasterNodeID(localStorage.getItem("masternodeID") ?? "");
```

Comment:
We should have different local storage key for different network so that we wont be having same masternode id for diff network

Reply:
updated to store into localStorage by context for masternodeID and rememberMasterNodeId checkbox